### PR TITLE
Prevent Command Prompt from opening with godcr on windows

### LINKED ISSUE #968: cli launches alongside godcr
While launching the godcr executable, a terminal window also opens up. This isn't the expected behaviour.

Expected behaviour is that godcr launches as a standalone app without needing to launch a terminal window.

zNNUBFePznZvzhIkQAfXeVFs

**Reported on WIndows OS**

### PR DESCRIPTION
**Pull request by devchoplife**
Closes #968 

This PR modifies the build process for windows to prevent the Command Prompt from opening with godcr on windows

The fix implemented was adding the `-H=windowsgui` flag to the windows build command. The tag hides the terminal and uses only the GUI.

This PR also updates the makefile and modifies incorrect links in the readme file

### CODE REVIEW

**Review by dreacot — APPROVED** [review]

File: makefile [hunk 1]
```diff
@@ -7,7 +7,7 @@ BUILD=`date -u +"%Y-%m-%dT%H:%M:%SZ"`
 BuildEnv="prod" 
 
 LDFLAGS=-ldflags "-w -s -X main.Version=${VERSION} -X main.BuildDate=${BUILD} -X main.BuildEnv=${BuildEnv}"
-
+LDFLAGSWIN= -ldflags "-H=windowsgui -w -s -X main.Version=${VERSION} -X main.BuildDate=${BUILD} -X main.BuildEnv=${BuildEnv}"
```
Comment (dreacot) [inline makefile:11]: add a comment here describing why this is necessary

**Review by dreacot — CHANGES_REQUESTED** [review]

File: makefile [hunk 2]
```diff
@@ -7,22 +7,26 @@ BUILD=`date -u +"%Y-%m-%dT%H:%M:%SZ"`
...
+all: clean macos windows linux freebsd
 
 freebsd:
-	GOOS=freebsd go build -trimpath ${LDFLAGS} -o ${BINARY}-freebsd-${GOARCH}
+	GOOS=freebsd GOARCH=amd64 go build -trimpath ${LDFLAGS} -o ${BINARY}-freebsd-${GOARCH}
+	GOOS=freebsd GOARCH=arm64 go build -trimpath ${LDFLAGS} -o ${BINARY}-freebsd-${GOARCH}
+
+linux:
+	GOOS=linux GOARCH=amd64 go build -trimpath ${LDFLAGS} -o ${BINARY}-freebsd-${GOARCH}
+	GOOS=linux GOARCH=arm64 go build -trimpath ${LDFLAGS} -o ${BINARY}-freebsd-${GOARCH}
```
Comment (Sirmorrison) [inline makefile]: does linux have same binary build as freebsd?

File: makefile [hunk 3]
```diff
@@ -17,8 +17,8 @@ freebsd:
 	GOOS=freebsd GOARCH=arm64 go build -trimpath ${LDFLAGS} -o ${BINARY}-freebsd-${GOARCH}
 
 linux:
-	GOOS=linux GOARCH=amd64 go build -trimpath ${LDFLAGS} -o ${BINARY}-freebsd-${GOARCH}
-	GOOS=linux GOARCH=arm64 go build -trimpath ${LDFLAGS} -o ${BINARY}-freebsd-${GOARCH}
+	GOOS=linux GOARCH=amd64 go build -trimpath ${LDFLAGS} -o ${BINARY}-linux-${GOARCH}
+	GOOS=linux GOARCH=arm64 go build -trimpath ${LDFLAGS} -o ${BINARY}-linux-${GOARCH}
 
 macos:
 	GOOS=darwin GOARCH=amd64 go build -trimpath ${LDFLAGS} -o ${BINARY}-darwin-${GOARCH}
```
Comment (dreacot) [inline makefile:24]: you don't seem to be targeting arm based machines

File: makefile [hunk 4]
```diff
@@ -7,22 +7,27 @@ BUILD=`date -u +"%Y-%m-%dT%H:%M:%SZ"`
...
+all: clean macos windows linux freebsd
 
 freebsd:
-	GOOS=freebsd go build -trimpath ${LDFLAGS} -o ${BINARY}-freebsd-${GOARCH}
+	GOOS=freebsd GOARCH=amd64 go build -trimpath ${LDFLAGS} -o ${BINARY}-freebsd-${GOARCH}
+	GOOS=freebsd GOARCH=arm64 go build -trimpath ${LDFLAGS} -o ${BINARY}-freebsd-${GOARCH}
+
+linux:
+	GOOS=linux GOARCH=amd64 go build -trimpath ${LDFLAGS} -o ${BINARY}-linux-${GOARCH}
+	GOOS=linux GOARCH=arm64 go build -trimpath ${LDFLAGS} -o ${BINARY}-linux-${GOARCH}
```
Comment (Sirmorrison) [inline makefile:21]: just out of curiosity, by do we have similar lines of codes repeated for all OS's? 
```
GOOS=freebsd GOARCH=amd64 go build -trimpath ${LDFLAGS} -o ${BINARY}-freebsd-${GOARCH}
GOOS=freebsd GOARCH=arm64 go build -trimpath ${LDFLAGS} -o ${BINARY}-freebsd-${GOARCH}
```
They are one and the same.

Comment (devchoplife) [reply makefile:21]: Yeah
The only change is the GOOS which is the OS we are building for and the GOARCH which is the architecture we are building for.

Every other things is constant except for windows with a different LDFLAG

**Review by Sirmorrison — DISMISSED** [review]

**Review by JustinBeBoy — DISMISSED** [review]

File: makefile [hunk 5]
```diff
@@ -7,22 +7,27 @@ BUILD=`date -u +"%Y-%m-%dT%H:%M:%SZ"`
...
 
 windows:
-	GOOS=windows go build -trimpath ${LDFLAGS} -o ${BINARY}-windows-${GOARCH}.exe
+	GOOS=windows GOARCH=amd64 go build -trimpath ${LDFLAGSWIN} -o ${BINARY}-windows-${GOARCH}.exe
  
 # Cleans our project: deletes old binaries
 clean:
 	-rm -f ${BINARY}-*
 
-.PHONY: clean darwin windows
```
Comment (dreacot) [inline makefile:28]: why was this deleted?

Comment (dreacot) [inline makefile:28]: why was this deleted?

**Review by dreacot — APPROVED** [review]
